### PR TITLE
Properly handle metadata updates

### PR DESCRIPTION
**Pull request by TheOneRing**
The first commit can be reviewed in https://github.com/owncloud/client/pull/8830

### CODE REVIEW

File: src/libsync/discovery.cpp [hunk 1]
```diff
@@ -1474,9 +1476,6 @@ void ProcessDirectoryJob::startAsyncLocalQuery()
         if (_serverQueryDone)
             this->process();
     });
-
-    QThreadPool *pool = QThreadPool::globalInstance();
-    pool->start(localJob); // QThreadPool takes ownership
```
Comment (dragotin) [inline src/libsync/discovery.cpp:1479]: Was this removed intentionally?

**Review by dragotin — APPROVED** [review]
LGTM, one question.

File: src/libsync/owncloudpropagator.cpp [hunk 2]
```diff
@@ -1112,4 +1130,43 @@ QString OwncloudPropagator::remotePath() const
...
+            FileSystem::setFileReadOnlyWeak(filePath, isReadOnly);
+            if (_item->_checksumHeader.isEmpty()) {
+                _item->_checksumHeader = prev._checksumHeader;
+            }
+            _item->_serverHasIgnoredFiles |= prev._serverHasIgnoredFiles;
+        }
+    }
+    const auto result = propagator()->updateMetadata(*_item);
+    if (!result) {
+        done(SyncFileItem::SoftError, tr("Could not update file : %1").arg(result.error()));
```
Comment (dragotin) [inline src/libsync/owncloudpropagator.cpp:1168]: In this case, the file is possibly treated by `setFileReadOnlyWeak` already, but the subsequent metadata update failed. Is it neccessary here to set the read-only flag back to be conform with the previous database entry?

**Review by dragotin — COMMENTED** [review]
I like this one, but one question.

File: src/libsync/discovery.cpp [hunk 3]
```diff
@@ -1047,23 +1050,20 @@ void ProcessDirectoryJob::processFileConflict(const SyncFileItemPtr &item, Proce
     if (up._valid && up._contentChecksum == serverEntry.checksumHeader) {
         // Solve the conflict into an upload, or nothing
         item->_instruction = up._modtime == localEntry.modtime && up._size == localEntry.size
-            ? CSYNC_INSTRUCTION_NONE : CSYNC_INSTRUCTION_SYNC;
+            ? CSYNC_INSTRUCTION_UPDATE_METADATA
+            : CSYNC_INSTRUCTION_SYNC;
```
[the remark this answers is not in the document]
Comment (fmoc) [reply src/libsync/discovery.cpp:1054]: This line looks like a `clang-format` victim.

File: src/libsync/discovery.cpp [hunk 4]
```diff
@@ -1047,23 +1050,20 @@ void ProcessDirectoryJob::processFileConflict(const SyncFileItemPtr &item, Proce
     if (up._valid && up._contentChecksum == serverEntry.checksumHeader) {
         // Solve the conflict into an upload, or nothing
```
Comment (fmoc) [inline src/libsync/discovery.cpp:1051]: Comment is probably outdated.

File: src/libsync/owncloudpropagator.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/libsync/owncloudpropagator.cpp:1168]: Hmm I assume having file permissions that differ from the db don't really hurt that much.
Also the step should be retried so once it succeeded we should be back in sync.

**Pull request merged**